### PR TITLE
Fix nightly tests

### LINKED ISSUE #429: Nightly build appears to be broken

### PR DESCRIPTION
**Pull request by Dr-Irv**
- [x] Closes #429 

Replaces PR #433, which was a starting point

Added tests to catch exceptions

Also avoid pyright 1.1.280 because it has bugs

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ types-pytz = ">= 2022.1.1"
 mypy = "0.990"
 pyarrow = ">=9.0.0"
 pytest = ">=7.1.2"
-pyright = ">=1.1.278"
+pyright = "<=1.1.279"
```
Comment (twoertwein) [inline pyproject.toml:40]: Maybe add a pyright issue number, if it exists

**Review by twoertwein — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (Dr-Irv) [reply pyproject.toml:40]: Latest main uses `pyright` >= 1.1.281, so this is resolved.

**Comment by Dr-Irv** [comment]
@bashtage  or @twoertwein  this is now good to go.  Can one of you merge?

**Comment by bashtage** [comment]
Good to have this in.

**Comment by bashtage** [comment]
Nightly will probably be breaking a lot in the next few months unfortunately.

**Pull request merged**